From 7220c973583d3c78ce4fb24e9a8d209a6ef08a56 Mon Sep 17 00:00:00 2001
From: Jeegar Patel <jeegar.patel@xilinx.com>
Date: Tue, 22 Sep 2020 05:05:26 -0700
Subject: [PATCH 1622/1851] drm: xlnx: bridge: Fix RESOURCE_LEAK type warning

commit 9466e75db8d9967ab9fbde6116d1b9d2e3794f3e from
https://github.com/Xilinx/linux-xlnx.git

This patch fix RESOURCE_LEAK type coverity warning.

Variable "cmd" going out of scope leaks the storage it points to.
Resource "cmd" is not freed or pointed-to in "strncpy_from_user".

Signed-off-by: Jeegar Patel <jeegar.patel@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/xlnx_bridge.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpu/drm/xlnx/xlnx_bridge.c b/drivers/gpu/drm/xlnx/xlnx_bridge.c
index 3cd0d83eac2e..7015beba3e6a 100644
--- a/drivers/gpu/drm/xlnx/xlnx_bridge.c
+++ b/drivers/gpu/drm/xlnx/xlnx_bridge.c
@@ -351,6 +351,7 @@ static ssize_t xlnx_bridge_debugfs_write(struct file *f, const char __user *buf,
 		if (ret < 0) {
 			pr_err("%s %d failed to copy the command  %s\n",
 			       __func__, __LINE__, buf);
+			kfree(cmd);
 			return ret;
 		}
 
-- 
2.31.1

